src/Teams/Services/ManageTeamsMembersService.cs

```diff
@@ -16,17 +17,27 @@ public class ManageTeamsMembersService : IManageTeamsMembersService
         private readonly IRepository<Team, int> _teamRepository;
         private readonly IRepository<TeamMember, int> _teamMemberRepository;
 
-        public ManageTeamsMembersService(IRepository<TeamMember, int> teamMemberRepository)
-        {
-            _teamMemberRepository = teamMemberRepository;
-        }
-
         public ManageTeamsMembersService(IRepository<Team, int> teamRepository, IRepository<TeamMember, int> memberRepository, ICurrentUser currentUser)
         {
             _currentUser = currentUser;
             _teamRepository = teamRepository;
             _teamMemberRepository = memberRepository;
         }
+        
+        public async Task<bool> RemoveAsync(int team_id, string member_id)
+        {
+            var team = await _teamRepository.GetAll().Include(t =>t.TeamMembers).Where(t => t.Id == team_id).FirstOrDefaultAsync();
+            if (team != null && team.TeamOwner == _currentUser.Current.Id() && MemberInTeam(team,member_id))
+            {       
+                return await _teamMemberRepository.DeleteAsync(team.TeamMembers.FirstOrDefault(t => t.MemberId == member_id));
+            }
+            return false;
+        }
+        
+        private bool MemberInTeam(Team team, string member_id)
```

Comment:
You are doing 2 iteration to check and get the team member, I think that it will much simpler to get the team member and if it's null, then no team member is presented, otherwise use founded record to delete it

Reply:
Sorry, but here https://github.com/epambrest/capacity/pull/41#discussion_r494587677 you wrote: But in this you need only to get bool value, AnyAsync will be much better for it.